### ISSUE TITLE
Scanner (src/scanner.cc) fails to build on MacOS

### ISSUE THREAD

**Issue by Hubro** [IndianBoy42/tree-sitter-just#23]
From the nvim-treesitter test pipeline:

```
Downloading tree-sitter-just...
Creating temporary directory
Extracting tree-sitter-just...
Compiling...
src/scanner.cc:15:24: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
  uint32_t prev_indent = 0;
                       ^
src/scanner.cc:47:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto curr = std::to_string(state->prev_indent);
  ^
src/scanner.cc:64:14: error: expected expression
    *state = {};
             ^
src/scanner.cc:88:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto advance = [lexer] { lexer->advance(lexer, false); };
  ^
src/scanner.cc:88:18: error: expected expression
  auto advance = [lexer] { lexer->advance(lexer, false); };
                 ^
src/scanner.cc:89:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto skip = [lexer] { lexer->advance(lexer, true); };
  ^
src/scanner.cc:89:15: error: expected expression
  auto skip = [lexer] { lexer->advance(lexer, true); };
              ^
src/scanner.cc:91:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto get_column = [lexer] { return lexer->get_column(lexer); };
  ^
src/scanner.cc:91:21: error: expected expression
  auto get_column = [lexer] { return lexer->get_column(lexer); };
                    ^
src/scanner.cc:128:5: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
    auto indent = get_column();
    ^
6 warnings and 4 errors generated.
Error detected while processing command line:
Error during compilation
Failed to execute the following command:
{
  cmd = "clang",
  err = "Error during compilation",
  info = "Compiling...",
  opts = {
    args = { "-o", "parser.so", "-I./src", "src/parser.c", "src/scanner.cc", "-Os", "-bundle", "-lstdc++", "-fPIC" },
    cwd = "/Users/runner/.local/share/nvim/tree-sitter-just"
  }
}
```

Ref: https://github.com/nvim-treesitter/nvim-treesitter/actions/runs/6935629490/job/18866152328?pr=5696#step:8:529

**Comment by 1995parham** [comment]
This thread helps me for building it on macOS.

**Comment by tgross35** [comment]
Same as #15, fixed in 88404a4c23756426786581d09e6432bcdeec8b7c
